lang/cs/Org.Apache.REEF.Network/Group/Task/Impl/OperatorTopology.cs

```diff
@@ -322,27 +305,7 @@ public T ReceiveFromChildren(IReduceFunction<T> reduceFunction)
                 throw new ArgumentNullException("reduceFunction");
             }
 
-            var receivedData = new List<T>();
-            var childrenToReceiveFrom = new HashSet<string>(_children.Select(node => node.Identifier));
-
-            while (childrenToReceiveFrom.Count > 0)
-            {
-                var childrenWithData = GetNodeWithData(childrenToReceiveFrom);
-
-                foreach (var child in childrenWithData)
-                {
-                    T[] data = ReceiveFromNode(child);
-                    if (data == null || data.Length != 1)
-                    {
-                        throw new InvalidOperationException("Received invalid data from child with id: " + child.Identifier);
-                    }
-
-                    receivedData.Add(data[0]);
-                    childrenToReceiveFrom.Remove(child.Identifier);
-                }
-            }
-
-            return reduceFunction.Reduce(receivedData);
+            return reduceFunction.Reduce(_childNodeContainer.GetDataFromAllChildren());
```

Comment:
@afchung  can you explain in a little bit of detail why there will be increased network overhead? I am ok addressing it in separate JIRA. This will hit user only if he does not use group communication in an optimal way.